tests/test-spawn-snid.lua

```diff
@@ -0,0 +1,44 @@
+--- Tests for spawn's startup notifications.
+
+local spawn = require("awful.spawn")
+
+local manage_called, c_snid
+
+client.connect_signal("manage", function(c)
+  manage_called = true
+  c_snid = c.startup_id
+  print("manage_called: c_snid:", tostring(c_snid))
```

Comment:
I will finish and merge https://github.com/awesomeWM/awesome/pull/533, and then rebase it on top of it.

I think while urxvt is OKish, gnome-terminal would proably pull in too many deps anyway.